### PR TITLE
[chainerx] Adds `log10`

### PR DESCRIPTION
**Pull request by ishanrai05**
Solves part of issue #6423

### CODE REVIEW

File: chainerx/_docs/routines.py [hunk 1]
```diff
@@ -1015,6 +1015,24 @@ def _docs_math():
...
+    _docs.set_doc(
+        chainerx.log10,
+        """log10(x)
+Base 10 logarithm, element-wise.
+
+Args:
+    x (~chainerx.ndarray): Input array.
+
+Returns:
+    :class:`~chainerx.ndarray`: Returned array: :math:`y = \\ln x`.
```
Comment (asi1024) [inline chainerx/_docs/routines.py]: Typo: ln -> log10

File: chainerx_cc/chainerx/routines/math.cc [hunk 2]
```diff
@@ -740,6 +740,27 @@ Array Log(const Array& x) {
...
+    {
+        NoBackpropModeScope scope{};
+        x.device().backend().CallKernel<Log10Kernel>(x, out);
+    }
+
+    BackwardBuilder bb{"log10", x, out};
+    if (BackwardBuilder::Target bt = bb.CreateTarget(0)) {
+        bt.Define([x_tok = bb.RetainInput(0)](BackwardContext& bctx) {
+            const Array& x = bctx.GetRetainedInput(x_tok);
+            bctx.input_grad() = *bctx.output_grad() / x;
```
Comment (asi1024) [inline chainerx_cc/chainerx/routines/math.cc]:
```suggestion
            bctx.input_grad() = *bctx.output_grad() / x / std::log(10);
```

File: chainerx/_docs/routines.py (hunk 1 above)
Comment (ishanrai05) [reply chainerx/_docs/routines.py]: Fixed it.

**Comment by asi1024** [comment]
Could you resolve conflicts?

File: chainerx_cc/chainerx/cuda/numeric.cuh [hunk 3]
```diff
@@ -41,6 +41,7 @@ __device__ inline cuda::Float16 Arccosh(cuda::Float16 x) { return cuda::Float16{
     }                                                           \
     __device__ inline cuda::Float16 name(cuda::Float16 x) { return cuda::Float16{func(static_cast<float>(x))}; }
 
+
```
Comment (asi1024) [inline chainerx_cc/chainerx/cuda/numeric.cuh]: It seems that you deleted `chainerx::Log10` when you merge.

File: chainerx_cc/chainerx/numeric.h [hunk 4]
```diff
@@ -53,4 +53,5 @@ CHAINERX_DEFINE_NATIVE_FLOAT16_FALLBACK_UNARY(Exp, std::exp)
 CHAINERX_DEFINE_NATIVE_FLOAT16_FALLBACK_UNARY(Log, std::log)
 CHAINERX_DEFINE_NATIVE_FLOAT16_FALLBACK_UNARY(Sqrt, std::sqrt)
 
+
```
Comment (asi1024) [inline chainerx_cc/chainerx/numeric.h]: Ditto.

Comment (ishanrai05) [reply chainerx_cc/chainerx/numeric.h]: Oh no...

Comment (ishanrai05) [reply chainerx_cc/chainerx/numeric.h]: I'll try to fix it

Comment (ishanrai05) [reply chainerx_cc/chainerx/numeric.h]: Done..

**Comment by asi1024** [comment]
Jenkins, test this please.

**Comment by asi1024** [comment]
Jenkins failure seems unrelated to this PR. LGTM!

**Pull request merged**